src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+    type = int)
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+
+    script_args = parser.parse_args()
+
+    return script_args
+
+filename = '''binary-ops-{NUMBER}.js'''
+MAX = (1<<53) - 1
+MIN = -1 * MAX
+BITMAX = 9999999
+BITMIN = -9999999
+
+class GenSequence:
+    def __init__(self, options, bit):
+        num_operands = options.operand_count[0]
+        self.num_operands = num_operands
+        self.operators_list = []
+        for i in range(len(options.op)):
+            if (bit == False):
+                if (options.op[i] == "+"):
+                    self.operators_list.append(self.add)
+                if (options.op[i] == "-"):
+                    self.operators_list.append(self.sub)
+                if (options.op[i] == "*"):
+                    self.operators_list.append(self.mul)
+                if (options.op[i] == "/"):
+                    self.operators_list.append(self.div)
+                if (options.op[i] == "%"):
+                    self.operators_list.append(self.mod)
+            else:
+                if (options.op[i] == "^"):
+                    self.operators_list.append(self.bitxor)
+        self.operands = [0] * num_operands
+        self.operators = [0] * (num_operands -1)
+
+        if (bit == False):
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(MIN, MAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(MIN, MAX)
+
+        else:
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(BITMIN, BITMAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(BITMIN, BITMAX)
+
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < MIN:
+            op2 = MIN - random.randint(MIN, op1)
+            sum = op1 + op2
+        elif sum > MAX:
+            op2 = MAX - random.randint(op1, MAX)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
+
+
+    def sub(self, op1, op2):
+        sum = op1 - op2
+        if sum < MIN:
+            op2 = - (MIN - random.randint(MIN, op1))
+            sum = op1 - op2
+        elif sum > MAX:
+            op2 = - (MAX - random.randint(op1, MAX))
+            sum = op1 - op2
+        return {'oper': '-', 'op2': op2, 'sum': sum}
+
+    def mul(self, op1, op2):
+        sum = op1 * op2
+        if sum < MIN:
+            op2 = MIN / op1
+        elif sum > MAX:
+            op2 = MAX / op1
+        if op2 > 0:
+            op2 = math.floor(op2)
+        else:
+            op2 = math.ceil(op2)
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    def div(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
```

Comment:
Where does this magic number come from?

Reply:
Replaced this with a random, that guaranteed not to be zero.